packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -3,70 +3,124 @@ pragma solidity >=0.8.0 <0.9.0;
+     * @param minInterestRate Minimum Annual percentage to be applied for loans using the lender's capital.
+     * @param collateralTokenAddress The address for the token contract that must be used to provide collateral for loans for this commitment.
+     * @param maxPrincipalPerCollateralAmount The amount of principal that can be used for a loan per each unit of collateral, expanded by 10^8. Use zero for no collateral required.
+     * @param collateralTokenType The type of asset of the collateralTokenAddres (ERC20, ERC721, or ERC1155).
+     * @param lender The address of the lender for this commitment.
+     * @param marketId The market id for this commitment.
+     * @param principalTokenAddress The address for the token contract that will be used to provide principal for loans of this commitment.
+     * @param borrower The address of the borrower that is allowed to accept this commitment.  Zero address is wildcard for any address. 
      */
     struct Commitment {
         uint256 maxPrincipal;
         uint32 expiration;
         uint32 maxDuration;
         uint16 minInterestRate;
+        address collateralTokenAddress;
+        uint256 collateralTokenId; 
+        uint256 maxPrincipalPerCollateralAmount; //zero means infinite . This is expressed using 16 decimals [PRINCIPAL_PER_COLLATERAL_EXPANSION_FACTOR] so 1 wei is 10^16 and 1 usdc is 10^22 
+        CommitmentCollateralType collateralTokenType; //erc721, erc1155 or erc20
+        address lender;
+        uint256 marketId;
+        address principalTokenAddress;
+        address borrower;
     }
-
-    modifier onlyMarketOwner(uint256 marketId) {
-        require(_msgSender() == getTellerV2MarketOwner(marketId));
-        _;
-    }
-
+ 
     // Mapping of lender address => market ID => lending token => commitment
     mapping(address => mapping(uint256 => mapping(address => Commitment)))
-        public lenderMarketCommitments;
+        public __lenderMarketCommitments_deprecated;
+
+    // CommitmentId => commitment
+    mapping(uint256 => Commitment) public lenderMarketCommitments;
+
+    uint256 commitmentCount;
 
     /**
-     * @notice This event is emitted when a lender's commitment is submitted.
+     * @notice This event is emitted when a lender's commitment is created.
      * @param lender The address of the lender.
      * @param marketId The Id of the market the commitment applies to.
      * @param lendingToken The address of the asset being committed.
      * @param tokenAmount The amount of the asset being committed.
      */
-    event UpdatedCommitment(
-        address indexed lender,
-        uint256 indexed marketId,
-        address indexed lendingToken,
-        uint256 tokenAmount
+    event CreatedCommitment(
+        uint256 indexed commitmentId,
+        address lender,
+        uint256 marketId,
+        address lendingToken,
+        uint256 tokenAmount,
+
+        address collateralToken 
```

Comment:
I don't think its necessary we put the collateral info in the event tbh - lets keep the event clean to future proof the event if we add functionality to have multiple collateral tokens
```suggestion
```